commands/poll.js

```diff
@@ -0,0 +1,333 @@
+        i === MAX_OPTIONS
+          ? "This is the last option of your poll!"
+          : `Additional option of the poll. You can add ${
+              MAX_OPTIONS - i
+            } more option/s after this!`
+      )
+      .setRequired(false)
+  );
+}
+
+pollSlashCommand.addIntegerOption((option) =>
+  option
+    .setName("close-in")
+    .setDescription(
+      "Default = 900 secs. Minimum = 300 secs. Polls have a 3-second buffer for calculating results."
+    )
+    .setMinValue(300)
+    .setRequired(false)
+);
+
+module.exports = {
+  data: pollSlashCommand,
+  async execute(interaction) {
+    console.debug("Creating poll...");
+
+    let pollEnded = false;
+    try {
+      const question = interaction.options.getString("question");
+      const pollCloseInSeconds =
+        interaction.options.getInteger("close-in") || 900;
+      const options = [];
+
+      for (let i = 1; i <= MAX_OPTIONS; ++i) {
+        const option = interaction.options.getString(`option-${i}`);
+
+        if (option) {
+          options.push({
+            emoji: `${OPTION_EMOJIS[i - 1]}`,
+            option: option,
+          });
+        }
+      }
+
+      const closePollUnixTimestamp =
+        Math.floor(Date.now() / 1000) + pollCloseInSeconds + DELAY;
+
+      const embed = {
+        color: 0xcf37ca,
+        title: `Poll Question: ${question}`,
+        description: `Created by ${interaction.member.user.tag} - poll expires <t:${closePollUnixTimestamp}:R>`,
+        fields: options.map((info) => {
+          return {
+            name: `${info.emoji}  - ${info.option}`,
+            value: "",
+          };
+        }),
+        footer: {
+          text: "This bot is powered by Xander's money",
+        },
+      };
+
+      const row = new ActionRowBuilder().addComponents(
+        new ButtonBuilder()
+          .setCustomId("close-poll")
+          .setLabel("Close Poll")
+          .setStyle(ButtonStyle.Danger)
+      );
+
+      const pollMessage = await interaction.reply({
+        embeds: [embed],
+        components: [row],
+        fetchReply: true,
+      });
+
+      for (let i = 0; i < options.length; ++i) {
+        await pollMessage.react(`${OPTION_EMOJIS[i]}`);
+      }
+
+      const reactionFilter = (reaction, user) => {
+        return OPTION_EMOJIS.includes(reaction.emoji.name) && !user.bot;
+      };
+
+      const reactionCollector = pollMessage.createReactionCollector({
+        reactionFilter,
+        time: pollCloseInSeconds * 1000,
+        dispose: true,
+      });
+
+      reactionCollector.on("collect", async (reaction, user) => {
+        console.info(`Got reaction ${reaction.emoji.name}  from ${user.tag}`);
+      });
+
+      reactionCollector.on("remove", async (reaction, user) => {
+        console.info(`Removed reaction ${reaction.emoji.name}  by ${user.tag}`);
+      });
+
+      reactionCollector.on("end", (collected) => {
+        console.debug(`Collected ${collected.size} reaction interactions!`);
+      });
```

Comment:
Query: Do we really need a reaction collector as of the moment? It seems that this is for logging purposes only, but I wouldn't really care to see what users the react or un-react to in the logs. Will this be part of a future enhancement (e.g. updating embed as someone reacts - although this might lag the bot if there are multiple polls/multiple users reacting at the same time) if ever?

Reply:
Okay, makes sense.